### PR TITLE
Split nodeset and deployment values

### PR DESCRIPTION
**Pull request by cjeanner**
#### Split nodeset and deployment values

To allow proper scale-out capabilities, we have to split the nodeset out
of the product deployment.

These templates prepare the field for the HCI split that will happen in
the architecture repository, before we split other VA/DT in the same
way.

#### Switch to names instead of ID

With this patch, we allow users to target a stage using a name instead
of a numerical ID.

This should ensure a more stable interface to the users in case we add
new stages in the automation scenario.

The "name" is built using the `build_output` basename, without the
extension.

For instance, if the build_output is "../control-plane.yaml", the stage
name will be `stage_control-plane` (but would also be accessible from
the ID).

Regarding injected hooks: since the run_hook role doesn't filter on
multiple strings (and will not allow it, at least for now), hooks are
still using the ID, and only the ID.

Depends-On: https://github.com/openstack-k8s-operators/architecture/pull/197

As a pull request owner and reviewers, I checked that:
- [X] Appropriate testing is done and actually running

### CODE REVIEW

**Comment by psathyan** [comment]
uni01alpha testing results

```
PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
compute-0                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
compute-1                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
localhost                  : ok=388  changed=120  unreachable=0    failed=0    skipped=141  rescued=1    ignored=0   

Thursday 25 April 2024  01:29:40 -0400 (0:00:00.186)       0:46:44.479 ******** 
=============================================================================== 
kustomize_deploy : Run Wait Conditions for examples/dt/uni01alpha ------------------------------------------------------------------------------------------------------------------------------------------------------------------- 768.64s
kustomize_deploy : Run Wait Conditions for examples/dt/uni01alpha/networker --------------------------------------------------------------------------------------------------------------------------------------------------------- 471.60s
kustomize_deploy : Run Wait Conditions for examples/dt/uni01alpha/control-plane ----------------------------------------------------------------------------------------------------------------------------------------------------- 450.02s
test_operator : Wait for the last job to be Completed - tempest --------------------------------------------------------------------------------------------------------------------------------------------------------------------- 396.85s
os_must_gather : Run openstack-must-gather command ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 185.09s
openshift_obs : Wait for observability-operator pod ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 60.61s
ci_setup : Install needed packages --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 54.03s
kustomize_deploy : Wait for cert-manager-operator pods ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 40.62s
kustomize_deploy : Wait for MetalLB speaker pods ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 36.04s
test_operator : Wait until the test-operator csv is present -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 33.56s
test_operator : Wait for the test-operator csv to Succeed ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 32.11s
openshift_setup : Wait for openshift-marketplace pods to be running ------------------------------------------------------------------------------------------------------------------------------------------------------------------ 31.83s
kustomize_deploy : Wait for certmanager pods ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 30.93s
kustomize_deploy : Wait for NMstate webhook deployment ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 20.75s
kustomize_deploy : Wait for NMstate handler pods ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 20.66s
artifacts : Ensure we have proper rights on the gathered content --------------------------------------------------------------------------------------------------------------------------------------------------------------------- 20.00s
kustomize_deploy : Wait for cainjector pods ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 11.00s
test_operator : Ensure that the test-operator-logs-pod is Running -------------------------------------------------------------------------------------------------------------------------------------------------------------------- 10.53s
kustomize_deploy : Run Wait Conditions for examples/dt/uni01alpha/control-plane/nncp -------------------------------------------------------------------------------------------------------------------------------------------------- 6.57s
kustomize_deploy : Wait for webhook pods ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 5.52s
```

File: roles/ci_gen_kustomize_values/templates/common/edpm-nodeset-values/values.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+{%   if _inst.startswith(_vm_type)                                                     %}
+{%     set _ = instances_names.append(_inst)                                           %}
+{%   endif                                                                             %}
+{% endfor                                                                              %}
+data:
+  ssh_keys:
+    authorized: {{ cifmw_ci_gen_kustomize_values_ssh_authorizedkeys | b64encode }}
+    private: {{ cifmw_ci_gen_kustomize_values_ssh_private_key | b64encode }}
+    public: {{ cifmw_ci_gen_kustomize_values_ssh_public_key | b64encode }}
+  nodeset:
```
[the remark this answers is not in the document]
Comment (cjeanner) [reply roles/ci_gen_kustomize_values/templates/common/edpm-nodeset-values/values.yaml.j2:17]: We didn't split all of the existing dataplane - so I'm keeping the "old" one in place for backward compatibility. We should be able to remove it once we moved all of the dataplane to the split format.

The "hci" monolithic edpm-values was used by some other DT, via symlinks (the ones I removed actually) - so moving it to the "common" directory sounds correct, to be able to merge "now" while preventing breaking too many things at the same time.

**Review by psathyan — APPROVED** [review]

**Comment by cjeanner** [comment]
/hold cancel

let's get it in.

**Pull request merged**